inst/include/child_model_simulation.hpp

```diff
@@ -13,54 +13,70 @@ void run_child_ageing(int time_step,
                       const State<ModelVariant, real_type> &state_curr,
                       State<ModelVariant, real_type> &state_next,
                       IntermediateData<ModelVariant, real_type> &intermediate) {
-  const auto demog = pars.base.demography;
-  const auto cpars = pars.children.children;
+  const auto survival_probability = pars.base.demography.survival_probability;
+  const auto hc_cd4_transition = pars.children.children.hc_cd4_transition;
+  const auto p_idx_fertility_first = pars.base.options.p_idx_fertility_first;
+
+  const auto hc1_hiv_pop_curr = state_curr.children.hc1_hiv_pop;
+  const auto hc1_art_pop_curr = state_curr.children.hc1_art_pop;
+  const auto hc2_hiv_pop_curr = state_curr.children.hc2_hiv_pop;
+  const auto hc2_art_pop_curr = state_curr.children.hc2_art_pop;
+
+  auto hc1_hiv_pop_next = state_next.children.hc1_hiv_pop;
+  auto hc1_art_pop_next = state_next.children.hc1_art_pop;
```

Comment:
I think we want to use `const auto&` and `auto&` here to ensure that these don't copy